### ISSUE TITLE
Resolving possible anomalies in SP800-53 catalog

### ISSUE THREAD

**Issue by wendellpiez** [usnistgov/OSCAL#218]
# User Story:

As an OSCAL user and SP800-53 user, I require confidence that the OSCAL SP800-53 catalog is complete and correct not only with respect to its content but also its structure.

## Goals:

Address some possible remaining (structural) anomalies in the OSCAL version of SP800-53. (These became apparent after data improvements in #175.)

## Dependencies:

Stable models. Clear decision-making authority over this representation of the standard.

## Acceptance Criteria

Possible anomalies in the data need to be examined and either corrected, or validated as already correct (or at any rate, adequate).

One such anomaly is the (lack of) structure under "Objectives". For example, in the AC-1 Objectives, we have labels "(a)(1)", "(a)(2)" "(b)(1)", "(b)(2)", but no labels "(a)" and "(b)" and no corresponding `part` elements for them. The same thing apparently occurs in a number of other controls (17 total: 
AC-1, AT-1, AU-1, CA-1, CM-1, CP-1, IA-1, IR-1, MA-1, MP-1, PE-1, PL-1, PS-1, RA-1, SA-1, SC-1, SI-1).

This Issue could also be staged to permit closer review of this data set, in case more such problems can be found. (Or, we could deal with this single question.)

- [ ] Is the data now correct with respect to any anomalies we have identified?
- [ ] Has the production pipeline for this content been amended so that hand-corrections do not need to be made (again) if/as this data set is regenerated from its (NVD XML) source?

**Comment by iMichaela** [comment]
## July 18 Status Update (Sprint 12 Acceptance Meeting) ##
The issue was not assigned to any member of the team in Sprint 12 and we will push the issue to Sprint 13

**Comment by wendellpiez** [comment]
Example of structural weaknesses in the OSCAL SP800-53 (reflecting similar lack of structure in NVD XML):

![image](https://user-images.githubusercontent.com/1588228/44468964-721e6d00-a5f4-11e8-83c5-4145b773510b.png)

There should be two `part` elements for "a" and "b" inside the "objectives" part, like so:

```
<part class="objective" id="ac-1_obj">
    <p>Determine if the organization:</p>
    <part class="objective" id="ac-1_obj_a">
        <part class="objective" id="ac-1_obj_a.1">...</part>
        <part class="objective" id="ac-1_obj_a.2">...</part>
    </part>
    <part class="objective" id="ac-1_obj_b">
        <part class="objective" id="ac-1_obj_b.1">...</part>
        <part class="objective" id="ac-1_obj_b.2">...</part>
    </part>
</part>
```

This better reflects the "intentional" (implicit) hierarchy of the data in the actual (explicit) hierarchy of the XML. The 'a' and 'b' branches are separated and given IDs (which will incidentally align with IDs elsewhere in the control description). The missing ID on the outside part is provided.

Since correct implicit structural IDs are impossible to derive from an incorrect structure, this is prerequisite to resolving the various ID issues. This implies the next step is XSLT to group the unstructured "objectives" elements. This should be possible using the labels in the NVD source (from which the present IDs are derived).

**Comment by wendellpiez** [comment]
**Sprint 13 Progress Aug 30 2018**

Tue Aug 28 we discussed this in the context also of #221. We agreed the solution would be to include a transformation step in the extraction pipeline to remediate the data where it lacks structure. I have written and tested an XSLT that does this, which is being deployed as part of the solution to #221 

So now the dependency is the other way. Before, #221 depended on this Issue; now this Issue is complete except for deployment, which awaits #221 

Progress towards goal: 90% (everything but deployment)

**Comment by wendellpiez** [comment]
** Sprint 13 Progress Report Sep 5 2018 **

As noted last week, nothing more to report: the improvements to the data extraction have been made and will show up with the acceptance of #221.

**Comment by david-waltermire-nist** [comment]
# 9/6/2018 Status Meeting

This is addressed in PR #229.

**Comment by iMichaela** [comment]
## 9/26/2018 ##
#229 was accepted and merged
